src/lib.rs

```diff
@@ -927,6 +927,13 @@ impl Connection {
     pub fn cache_flush(&self) -> Result<()> {
         self.db.borrow_mut().cache_flush()
     }
+
+    /// Determine if a database is read-only
+    #[cfg(feature = "modern_sqlite")] // 3.7.11
+    #[cfg_attr(docsrs, doc(cfg(feature = "modern_sqlite")))]
+    pub fn db_readonly(&self, db_name: DatabaseName<'_>) -> Result<bool> {
```

Comment:
nit: maybe `is_readonly()` is a better name for the public API?

Reply:
You are right.